src/nodes/typeGuards/index.ts

```diff
@@ -0,0 +1,3 @@
+export * from "./compound";
+export * from "./single";
+export * from "./union";
```

Comment:
This is needed for the published ESM output. I've learned this the hard way...

```suggestion
export * from "./compound.js";
export * from "./single.js";
export * from "./union.js";
```

Not great that this didn't fail any build. 

- https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues/233
- https://github.com/JoshuaKGoldberg/ts-api-utils/issues/35
- https://github.com/JoshuaKGoldberg/ts-api-utils/pull/36

Reply:
There are other places where there are no extensions on the main branch that would need to be fix.
We should probably just bundle esm https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues/233#issuecomment-1423589494

Reply:
Hmm yeah they are broken 😂. Interesting idea!